### PR TITLE
add missing implementation for casbin@0.3.0 and make unique key constraint works

### PR DESCRIPTION
**Pull request by GopherJ**
Hello, this PR make unique key works for policy rules, before it cannot work because we do have nullable fields.

Now we are adding empty string to solve this issue.

Besides, it adds implementation of `add_policies` and `remove_policies` using diesel transaction.

### CODE REVIEW

**Comment by GopherJ** [comment]
@hsluoyz @omid could you review?

**Comment by hsluoyz** [comment]
@GopherJ why don't use the built-in services provided by Travis CI? Like how we did it in Xorm adapter: https://github.com/casbin/xorm-adapter/blob/79a2aa54a016320eb29cf90090f642183827750b/.travis.yml#L14-L16

**Comment by GopherJ** [comment]
hello @hsluoyz we are using it, but in dev I would like to have some scripts so I added these two scripts

**Comment by hsluoyz** [comment]
Got it.

File: src/adapter.rs [hunk 1]
```diff
@@ -188,16 +201,46 @@ impl Adapter for DieselAdapter {
...
+            .get()
+            .map_err(|err| Box::new(Error::PoolError(err)) as Box<dyn StdError>)?;
+
+        let new_rules = rules
+            .into_iter()
+            .filter_map(|x: Vec<&str>| self.save_policy_line(ptype, x))
+            .collect::<Vec<NewCasbinRule>>();
+
+        conn.transaction::<_, DieselError, _>(|| {
+            diesel::insert_into(casbin_rules)
```
Comment (omid) [inline src/adapter.rs]: Do we need the transaction here? I think Diesel will generate one query.

Comment (GopherJ) [reply src/adapter.rs]: I was also thinking about this. But I'm not really sure about it. I'm not sure it's one query or multiple. If it's one query we can then remove it.

Comment (omid) [reply src/adapter.rs]: I'm also not 100% sure in every case :see_no_evil:

**Pull request merged**